docs/content/en/docs/reference/clusterspec/cloudstack.md

```diff
@@ -5,4 +5,346 @@ weight: 30
+
+>**_NOTE:_** The taints provided will be used instead of the default control plane taint `node-role.kubernetes.io/master`.
+Any pods that you run on the control plane nodes must tolerate the taints you provide in the control plane configuration.
+>
+
+### controlPlaneConfiguration.labels
+A list of labels to apply to the control plane nodes of the cluster. This is in addition to the labels that
+EKS Anywhere will add by default.
+For example, this label identifies a Cluster API failuredomain, which causes CAPC to substitute the value for the actual AZ name as a label on the nodes:
+
+```yaml
+    labels:
+      cluster.x-k8s.io/failure-domain: ds.meta_data.failuredomain
+```
+
+Modifying the labels associated with the control plane configuration will cause new nodes to be rolled out, replacing
+the existing nodes.
+
+### datacenterRef
+Refers to the Kubernetes object with CloudStack environment specific configuration. See `CloudStackDatacenterConfig Fields` below.
+
+### externalEtcdConfiguration.count
+Number of etcd members
+
+### externalEtcdConfiguration.machineGroupRef
+Refers to the Kubernetes object with CloudStack specific configuration for your etcd members. See `CloudStackMachineConfig Fields` below.
+
+### kubernetesVersion (required)
+The Kubernetes version you want to use for your cluster. Supported values: `1.23`, `1.22`, `1.21`, `1.20`
+
+### managementCluster (required)
+Identifies the name of the management cluster.
+If this is a standalone cluster or if it were serving as the management cluster for other workload clusters, this will be the same as the cluster name.
+
+### workerNodeGroupConfigurations (required)
+This takes in a list of node groups that you can define for your workers.
+You may define one or more worker node groups.
+
+### workerNodeGroupConfigurations.count (required)
+Number of worker nodes
+
+### workerNodeGroupConfigurations.machineGroupRef (required)
+Refers to the Kubernetes object with CloudStack specific configuration for your nodes. See `CloudStackMachineConfig Fields` below.
+
+### workerNodeGroupConfigurations.name (required)
+Name of the worker node group (default: md-0)
+
+### workerNodeGroupConfigurations.taints
+A list of taints to apply to the nodes in the worker node group.
+
+Modifying the taints associated with a worker node group configuration will cause new nodes to be rolled-out, replacing the existing nodes associated with the configuration.
+
+At least one node group must not have `NoSchedule` or `NoExecute` taints applied to it.
+
+### workerNodeGroupConfigurations.labels
+A list of labels to apply to the nodes in the worker node group. This is in addition to the labels that
+EKS Anywhere will add by default.
+For example, this label identifies a Cluster API failuredomain, which causes CAPC to substitute the value for the actual AZ name as a label on the nodes:
```

Comment:
I'm not sure if "for example" makes sense here, since this keyword `ds.meta_data.failuredomain` is specifically reserved for the availability zones replacement usecase. It's more of a custom feature than a generic example

Reply:
Okay. I think it's the only custom label as well. I'll remove "For example".

Reply:
I reworded base on comments above.

Reply:
Customer can add other labels like `foo: bar` and nodes will be labeled like that.

`ds.meta_data.failuredomain` is a special keyword case Cloudstack supported.